blog: Neo usage limits launch post (#20203)
pulumi/docs: content/blog/neo-usage-limits/index.md

---
title: "Introducing Usage Limits for Pulumi Neo"
date: 2026-07-13T08:00:00-07:00
draft: true

> meta_desc: "Pulumi Neo usage limits set a dollar boundary on monthly Neo spend. Neo pauses when it reaches your organization or per-member limit, with email alerts."

Reviewer: ```suggestion
meta_desc: "Set monthly organization or per-member Pulumi Neo dollar limits and get alerts via email.."
```

feature_image: feature.png
authors:
    - john-keiser
tags:
    - pulumi-neo
    - ai
    - ai-agents
    - features
category: product
schema_type: auto

# canonical_url: set to /docs/ai/usage-limits/ once that docs page is live (not merged yet).

social:
    twitter: "Now Pulumi Neo supports monthly usage limits, so you can hand it more infrastructure work, without worrying about the bill."
    linkedin: "Introducing usage limits for Pulumi Neo. Set a monthly dollar limit on Neo usage, for your whole organization or per member, so you can hand it more infrastructure work without worrying about the bill. Neo pauses when spend reaches the limit, and admins get email alerts as usage climbs."
    bluesky: "Now Pulumi Neo supports monthly usage limits, so you can hand it more infrastructure work, without worrying about the bill."

> ---

Reviewer: There is too much "Worrying" about the bill here. 

It should be positioned more as the control we give to organizations. let's update throughout the blog post.

Author: Good idea. Switched to positive framing (control) instead of negative framing around avoiding worry.

> When you use [Pulumi Neo](/docs/ai/) to take on infrastructure work, it's natural to want to hand it more and more. A monthly usage limit lets you do exactly that, without worrying about the bill: set a dollar limit on Neo's spend, and it pauses when your organization reaches the limit instead of running past it.

Reviewer: Prompt Claude to use less of "instead of". Stating that Neo is paused only its reach is good.

I would add a quick intro paragraph for Neo here.

Author: I put the brief Neo summary into the first sentence--let me know if too brief :)

<!--more-->

## How usage limits work

> Your organization limit is a single monthly dollar amount covering all Neo usage across the org. You set it in Pulumi Cloud. When usage reaches the limit, Neo pauses for the rest of the billing period and resumes automatically at the start of the next one. An Admin or Billing Manager can raise the limit to resume before then.

Reviewer: Can we add the steps on how they can set this in the console?

![The Manage token usage panel, where an admin sets the organization's monthly Neo limit and turns on email notifications.](manage-token-usage.png)

> Enforcement happens at a natural boundary in Neo's work, so a task already in progress finishes its current step before pausing. A limit can therefore land a few dollars over, which is expected rather than a billing error.

Reviewer: ```suggestion
Enforcement happens at a natural boundary in Neo's work, so a task already in progress finishes its current step before pausing. Therefore, usage could go a few dollars over the set limit. 
.
```

## Per-member limits and alerts

> You can also set separate limits for each individual member, which pauses Neo for them after they hit the limit, without affecting others. (The organization limit still applies, no matter what.)

Reviewer: Remove "wihtout affecting others"

We should state clearly that when both user limits and organization limits are set, which one would apply in what scenario - just as a example.

---
title: "Introducing Usage Limits for Pulumi Neo"

> date: 2026-07-13T08:00:00-07:00

Reviewer 2: ```suggestion
date: 2026-07-14
```

draft: false
meta_desc: "Set monthly organization or per-member Pulumi Neo dollar limits and get alerts via email."
feature_image: feature.png
authors:
    - john-keiser
tags:
    - pulumi-neo
    - ai
    - ai-agents
    - features
category: product
schema_type: auto

# canonical_url: set to /docs/ai/usage-limits/ before publish (docs page is live).

> 

Reviewer 2: I'm assuming we don't want to do this, since the docs and blog are different, and we want both indexed. Right?

```suggestion
```
